yarn.lock

```diff
@@ -4418,16 +4418,11 @@ lodash.flatmap@^4.5.0:
   resolved "https://registry.yarnpkg.com/lodash.flatmap/-/lodash.flatmap-4.5.0.tgz#ef8cbf408f6e48268663345305c6acc0b778702e"
   integrity sha1-74y/QI9uSCaGYzRTBcaswLd4cC4=
 
-lodash@4.17.15, lodash@^4.17.13, lodash@^4.17.14, lodash@^4.17.15, lodash@^4.17.4:
+lodash@4.17.15, lodash@=4.17.11, lodash@^4.17.12, lodash@^4.17.13, lodash@^4.17.14, lodash@^4.17.15, lodash@^4.17.4:
```

Comment:
I'm not familiar with resolutions or dependencies in general, but it looks like `lodash@=4.17.11` is still there?

Reply:
Yes. I don't understand what happened here either. I think these are all the versions that were requested from sub dependencies, and immediately below shows the resolution value that was chosen. I was just trying to fix the issue without manually editing the lock file. I wonder if @nlordell has a suggestion on how to do this better, or what happened here.

Reply:
```
lodash@4.17.15, lodash@=4.17.11, lodash@^4.17.12, lodash@^4.17.13, lodash@^4.17.14, lodash@^4.17.15, lodash@^4.17.4:
  version "4.17.15"
  resolved "https://registry.yarnpkg.com/lodash/-/lodash-4.17.15.tgz#b447f6670a0455bbfeedd11392eff330ea097548"
```

From what I understand of the lock file syntax, it is basically saying that all `package.json` dependencies that request the versions `4.17.15, =4.17.11, ^4.17.12, ^4.17.13, ^4.17.14, ^4.17.15, ^4.17.4` resolve to the actual version `4.17.15`.

When I read this, to me it suggests that only `4.17.15` is actually being used, and that there are 7 different requested versions that resolve to this.

Reply:
Does it mean that when you set a "resolution" in your package.json it might make yarn ignore strict requirements like `lodash@=4.17.11`?
So, if somebody used resolutions there'd be the chance of using broken packages because they'd use wrong dependencies?
Or maybe `@=` doesn't actually mean equal and there is some leeway?

Reply:
> Does it mean that when you set a "resolution" in your package.json it might make yarn ignore strict requirements like lodash@=4.17.11?

From what I understand that is precisely [a use case of `resolution`s](https://classic.yarnpkg.com/en/docs/selective-version-resolutions/#toc-why-would-you-want-to-do-this).

> So, if somebody used resolutions there'd be the chance of using broken packages because they'd use wrong dependencies?

Yes, I think that is why `resolution`s should be used sparingly. I think patching security vulnerabilities in transient dependencies in an appropriate use for them (obviously, some sort of testing is required to ensure that this does not break anything).

> Or maybe @= doesn't actually mean equal and there is some leeway?

As far as I know `=VERSION` specs mean exact matches.

Reply:
Scary, I thought they'd try to enforce a single version of the package as long as the prerequisites were met. Very good to know that this is not the case!

Reply:
Im actually not sure what it does in that situation. I’d imagine it would try to find resolve multiple version requests with as few resolved (as in in the lock file, not in the `resolve` section) versions as possible.

Note that here it’s not possible because `=4.17.11` and `^4.17.12` have no overlap